airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryStagingConsumerFactory.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.bigquery;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.base.Functions;
+import com.google.common.base.Preconditions;
+import io.airbyte.commons.concurrency.VoidCallable;
+import io.airbyte.commons.functional.CheckedBiConsumer;
+import io.airbyte.commons.functional.CheckedBiFunction;
+import io.airbyte.commons.functional.CheckedConsumer;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.integrations.base.AirbyteMessageConsumer;
+import io.airbyte.integrations.base.AirbyteStreamNameNamespacePair;
+import io.airbyte.integrations.destination.bigquery.formatter.BigQueryRecordFormatter;
+import io.airbyte.integrations.destination.buffered_stream_consumer.BufferedStreamConsumer;
+import io.airbyte.integrations.destination.record_buffer.SerializableBuffer;
+import io.airbyte.integrations.destination.record_buffer.SerializedBufferingStrategy;
+import io.airbyte.protocol.models.AirbyteMessage;
+import io.airbyte.protocol.models.AirbyteStream;
+import io.airbyte.protocol.models.ConfiguredAirbyteCatalog;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class BigQueryStagingConsumerFactory {
```

Comment:
Would it be possible to get a common staging consumer factory from snowflake and bigquery?

we probably don't want to have their own staging consumer factory for each destination moving forward, no? (redshift, etc)

Reply:
I did try to reuse the existing factory or modify the Snowflake factory to fit in BigQuery. However, it complicates the factory a lot, because BigQuery is different and it has its own client. I don't think we need another consumer factory for destinations like Redshift, because most destinations are JDBC based, and fit into the current `StagingConsumerFactory`.